src/components/Files/ExcalidrawPage.tsx

```diff
@@ -0,0 +1,185 @@
+import { Excalidraw } from "@excalidraw/excalidraw";
+import { type ExcalidrawElement } from "@excalidraw/excalidraw/types/element/types";
+import { useMutation, useQuery } from "@tanstack/react-query";
+import { t } from "i18next";
+import { navigate } from "raviger";
+import { useEffect, useState } from "react";
+import { toast } from "sonner";
+
+import { debounce } from "@/lib/utils";
+
+import CareIcon from "@/CAREUI/icons/CareIcon";
+
+import { Button } from "@/components/ui/button";
+import { Input } from "@/components/ui/input";
+
+import Loading from "@/components/Common/Loading";
+import { CreateFileResponse } from "@/components/Patient/models";
+
+import routes from "@/Utils/request/api";
+import mutate from "@/Utils/request/mutate";
+import query from "@/Utils/request/query";
+
+type Props = {
+  associatingId: string;
+  fileType: string;
+  drawingId?: string;
+};
+
+export default function ExcalidrawPage({
+  associatingId,
+  drawingId,
+  fileType,
+}: Props) {
+  const [elements, setElements] = useState<readonly ExcalidrawElement[] | null>(
+    drawingId ? null : [],
+  );
+  const [name, setName] = useState("");
+  const [id, setId] = useState(drawingId);
+  const [isDirty, setIsDirty] = useState(false);
+
+  const { data, isLoading } = useQuery({
+    queryKey: ["file", id],
+    queryFn: query(routes.retrieveUpload, {
+      pathParams: { id: id || "" },
+    }),
+    enabled: !!id,
+  });
```

Comment:
if `ExcalidrawPage` component can act as both viewer and editor, why do we need a separate component `ExcalidrawView`?